### PR TITLE
feat(frontend): add community switch to send coins

### LINKED ISSUE #2948: 🚀 [Feature] X-SendCoins 1: extend sendcoin-page with community selection field
## 🚀 Feature

- [x] new selection component for community
- [x] default selection is home-community with attribute `foreign = false`
- [x] show in the selectionlist the name of a community
- [x] get list of Communities from backend by invoking communities, which will only return verified communities
- [x] from the selected community the attribute id/url or later the uuid could be used as identifier for the sendCoins invokation

### PR DESCRIPTION
**Pull request by einhornimmond**
## 🍰 Pullrequest

- add community switch component
- use it in send coins
- show the choosen target community in confirm transaction component
- add target community id to mutation call
- update backend mutation add target community id as parameter (currently not used)

### Issues

- fixes #2948

### Todo

- [x] Add Test for CommunitySwitch
- [x] Update Tests

### CODE REVIEW

File: frontend/src/components/GddSend/TransactionForm.vue [hunk 1]
```diff
@@ -147,24 +153,26 @@ export default {
...
     InputTextarea,
+    CommunitySwitch,
   },
   props: {
     balance: { type: Number, default: 0 },
     identifier: { type: String, default: '' },
     amount: { type: Number, default: 0 },
     memo: { type: String, default: '' },
     selected: { type: String, default: 'send' },
+    targetCommunity: { type: Object, default: { id: 0, name: COMMUNITY_NAME } },
```
Comment (clauspeterhuebner) [inline frontend/src/components/GddSend/TransactionForm.vue:164]: is this correct to set the default selected community from configuration than from the previous read community-list?

File: frontend/src/components/GddSend/TransactionForm.vue [hunk 2]
```diff
@@ -179,13 +187,15 @@ export default {
...
         userName: this.userName,
+        targetCommunity: this.form.targetCommunity,
       })
     },
     onReset(event) {
       event.preventDefault()
       this.form.identifier = ''
       this.form.amount = ''
       this.form.memo = ''
+      this.form.targetCommunity = { id: 0, name: COMMUNITY_NAME }
```
Comment (clauspeterhuebner) [inline frontend/src/components/GddSend/TransactionForm.vue:198]: is this correct to set the default selected community from configuration than from the previous read community-list?

File: frontend/src/graphql/mutations.js [hunk 3]
```diff
@@ -71,8 +71,18 @@ export const createUser = gql`
...
+    $identifier: String!
+    $amount: Decimal!
+    $memo: String!
+    $targetCommunity: Int!
+  ) {
+    sendCoins(
+      identifier: $identifier,
+      amount: $amount,
+      memo: $memo,
+      targetCommunity: $targetCommunity
```
Comment (clauspeterhuebner) [inline frontend/src/graphql/mutations.js:84]: i would define the targetCommunity identifier as the 1st parameter, perhaps would it rename to recipientCommunityIdentifier and would rename the 2nd parameter to recipientIdentifier

File: frontend/src/graphql/queries.js
Comment (clauspeterhuebner) [inline frontend/src/graphql/queries.js:1]: i don't know on which base you start with your changes, but i miss the following queries, which are still on master:

export const communityInfo = gql`
  query {
    getCommunityInfo {
      name
      description
      registerUrl
      url
    }
  }
`

export const communities = gql`
  query {
    communities {
      id
      name
      url
      description
      registerUrl
    }
  }
`

export const getCommunities = gql`
  query {
    getCommunities {
      id
      foreign
      publicKey
      url
      lastAnnouncedAt
      verifiedAt
      lastErrorAt
      createdAt
      updatedAt
    }
  }
`

export const getCommunitySelections = gql`
  query {
    getCommunitySelections {
      id
      foreign
      name
      description
      url
      creationDate
      uuid
      authenticatedAt
    }
  }
`

**Review by clauspeterhuebner — CHANGES_REQUESTED** [review]

File: backend/src/graphql/resolver/CommunityResolver.ts [hunk 4]
```diff
@@ -26,7 +26,7 @@ export class CommunityResolver {
 
   @Authorized([RIGHTS.COMMUNITIES])
   @Query(() => [Community])
-  async getCommunitySelections(): Promise<Community[]> {
+  async communities(): Promise<Community[]> {
```
[the remark this answers is not in the document]
Comment (einhornimmond) [reply backend/src/graphql/resolver/CommunityResolver.ts:29]: We have talked about it. If we take the overhead of using graphql, we can use it properly. This resolver is a standard resolver it just returns the list of all communities. If you go by the naming pattern of other projects that use graphql, this is what you would call it. 
I want to refactor the use of graphql piece by piece, always where I'm at anyway.

[the remark this answers is not in the document]
Comment (einhornimmond) [reply backend/src/graphql/resolver/CommunityResolver.ts:29]: The resolver for the table federated_communities for returning the list would be named federatedCommunities.

File: frontend/src/components/CommunitySwitch.vue [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  methods: {
+    updateCommunity(community) {
+      this.value = community
+      this.$emit('input', this.value)
+    },
+    setDefaultCommunity() {
+      // set default community, the only one which isn't foreign
+      // we assume it is only one entry with foreign = false
+      if (!this.value.id && this.communities.length) {
+        const foundCommunity = this.communities.find((community) => !community.foreign)
```
[the remark this answers is not in the document]
Comment (einhornimmond) [reply frontend/src/components/CommunitySwitch.vue:39]: I see in the CommunityResolver that the returned Community-Array is ordered by name, this is the code:
```ts
@Authorized([RIGHTS.COMMUNITIES])
  @Query(() => [Community])
  async communities(): Promise<Community[]> {
    const dbCommunities: DbCommunity[] = await DbCommunity.find({
      order: {
        name: 'ASC',
      },
    })
    return dbCommunities.map((dbCom: DbCommunity) => new Community(dbCom))
  }
  ```

File: frontend/src/components/GddSend/TransactionConfirmationSend.vue [hunk 6]
```diff
@@ -76,11 +76,11 @@ export default {
     amount: { type: Number, required: true },
     memo: { type: String, required: true },
     userName: { type: String, default: '' },
+    targetCommunity: { type: Object, default: { id: 0, name: COMMUNITY_NAME }},
```
[the remark this answers is not in the document]
Comment (einhornimmond) [reply frontend/src/components/GddSend/TransactionConfirmationSend.vue:79]: The CommunitySwitch component reads the community list and is the only one that can access it unless I pass it on. But I see no reason for that. Outside the select component I only need the choosen community.

File: frontend/src/components/GddSend/TransactionForm.vue (hunk 1 above)
Comment (einhornimmond) [reply frontend/src/components/GddSend/TransactionForm.vue:164]: The CommunitySwitch component reads the community list and is the only one that can access it unless I pass it on. But I see no reason for that. Outside the select component I only need the choosen community.

File: frontend/src/graphql/mutations.js (hunk 3 above)
Comment (einhornimmond) [reply frontend/src/graphql/mutations.js:84]: ok

File: frontend/src/graphql/queries.js
Comment (einhornimmond) [reply frontend/src/graphql/queries.js:1]: Are you sure that you find them in [frontend/src/graphql/queries.js](https://github.com/einhornimmond/gradido/blob/master/frontend/src/graphql/queries.js)?
They are in [backend/src/seeds/graphql/queries.ts](https://github.com/gradido/gradido/blob/master/backend/src/seeds/graphql/queries.ts)

But they are currently not used. I have now removed the query for which no resolver exist and update the existing alongside the change I had made in the resolver.

File: frontend/src/components/GddSend/TransactionForm.vue (hunk 2 above)
Comment (einhornimmond) [reply frontend/src/components/GddSend/TransactionForm.vue]: The CommunitySwitch component reads the community list and is the only one that can access it unless I pass it on. But I see no reason for that. Outside the select component I only need the choosen community.

File: backend/src/graphql/arg/TransactionSendArgs.ts [hunk 7]
```diff
@@ -19,4 +19,7 @@ export class TransactionSendArgs {
   @MaxLength(MEMO_MAX_CHARS)
   @MinLength(MEMO_MIN_CHARS)
   memo: string
+
+  @Field(() => Int)
+  targetCommunity: number
```
[the remark this answers is not in the document]
Comment (einhornimmond) [reply backend/src/graphql/arg/TransactionSendArgs.ts]: erledigt

File: frontend/src/components/CommunitySwitch.vue (hunk 5 above)
[the remark this answers is not in the document]
Comment (clauspeterhuebner) [reply frontend/src/components/CommunitySwitch.vue:43]: ah ja da hast du völlig Recht

**Review by clauspeterhuebner — APPROVED** [review]

**Review by clauspeterhuebner — APPROVED** [review]

**Pull request merged**